python/ray/tune/progress_reporter.py

```diff
@@ -0,0 +1,163 @@
+                    "unexpected crashes. Consider "
+                    "reducing the memory used by your application "
+                    "or reducing the Ray object store size by setting "
+                    "`object_store_memory` when calling `ray.init`.")
+        else:
+            warn = ""
+        return "Memory usage on this node: {}/{} GiB{}".format(
+            round(used_gb, 1), round(total_gb, 1), warn)
+    except ImportError:
+        return ("Unknown memory usage. Please run `pip install psutil` "
+                "(or ray[debug]) to resolve)")
+
+
+def trial_progress_str(trials, metrics=None, fmt="psql", max_rows=100):
+    """Returns a human readable message for printing to the console.
+
+    This contains a table where each row represents a trial, its parameters
+    and the current values of its metrics.
+
+    Args:
+        trials (List[Trial]): List of trials to get progress string for.
+        metrics (List[str]): Names of metrics to include. Defaults to
+            metrics defined in DEFAULT_RESULT_KEYS.
+        fmt (str): Output format (see tablefmt in tabulate API).
+        max_rows (int): Maximum number of rows in the trial table.
+    """
+    messages = []
+    delim = "<br>" if fmt == "html" else "\n"
+    if len(trials) < 1:
+        return delim.join(messages)
+
+    num_trials = len(trials)
+    trials_per_state = {}
+    for t in trials:
+        trials_per_state[t.status] = trials_per_state.get(t.status, 0) + 1
+    messages.append("Number of trials: {} ({})".format(num_trials,
+                                                       trials_per_state))
+
+    if num_trials > max_rows:
+        overflow = num_trials - max_rows
+        # TODO(ujvl): suggestion for users to view more rows.
+        messages.append("Table truncated to {} rows ({} overflow).".format(
+            max_rows, overflow))
+
+    # Pre-process trials to figure out what columns to show.
+    keys = list(metrics or DEFAULT_PROGRESS_KEYS)
+    keys = [k for k in keys if any(t.last_result.get(k) for t in trials)]
+    has_failed = any(t.error_file for t in trials)
+    # Build rows.
+    trial_table = []
+    params = list(set().union(*[t.evaluated_params for t in trials]))
+    for trial in trials[:min(num_trials, max_rows)]:
+        trial_table.append(_get_trial_info(trial, params, keys, has_failed))
+    # Parse columns.
+    parsed_columns = [REPORTED_REPRESENTATIONS.get(k, k) for k in keys]
+    columns = ["Trial ID", "status", "loc"]
+    columns += ["failures", "error file"] if has_failed else []
+    columns += params + parsed_columns
+    messages.append(
+        tabulate(trial_table, headers=columns, tablefmt=fmt, showindex=False))
+    return delim.join(messages)
+
+
+def _get_trial_info(trial, parameters, metrics, include_error_data=False):
+    """Returns the following information about a trial:
+
+    ID | status | loc | # failures | error_file | params ... | metrics ...
```

Comment:
in followup to the previous comment, maybe this can be 
```
trial_name | ID | status | loc | # failures | error_file | params ... | metrics ...
```

This is just to preserve backwards compat for now.

Reply:
The corresponding thing would just be `str(trial)`.

Reply:
I can include it but it'll for sure typically make each row longer than screen width since it also contains the params. 

should i remove the parameter columns for now?

Reply:
What if you made
1. the experiment tag not part of the default trial name and instead did `{str(trial)}_{trial_id}` instead of `{str(trial)}_{experiment_tag}`

2. Added the `experiment_tag` to `trial.update_last_result` as `result.update(experiment_tag=self.experiment_tag)`.

This way, we can 1. keep the trial name while conserving space and 2. allow the user still to access the experiment tag.